staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt-config.go

```diff
@@ -0,0 +1,246 @@
+	pli.Informer().AddEventHandler(reqMgmt)
+	fsi.Informer().AddEventHandler(reqMgmt)
+}
+
+func (reqMgmt *requestManagementSystem) triggerReload() {
+	klog.V(7).Info("request-management system reloading")
+	reqMgmt.configQueue.Add(0)
+}
+
+// OnAdd handles notification from an informer of object creation
+func (reqMgmt *requestManagementSystem) OnAdd(obj interface{}) {
+	reqMgmt.triggerReload()
+}
+
+// OnUpdate handles notification from an informer of object update
+func (reqMgmt *requestManagementSystem) OnUpdate(oldObj, newObj interface{}) {
+	reqMgmt.triggerReload()
+}
+
+// OnDelete handles notification from an informer of object deletion
+func (reqMgmt *requestManagementSystem) OnDelete(obj interface{}) {
+	reqMgmt.triggerReload()
+}
+
+func (reqMgmt *requestManagementSystem) Run(stopCh <-chan struct{}) error {
+	defer reqMgmt.configQueue.ShutDown()
+	klog.Info("Starting reqmgmt config controller")
+	if ok := cache.WaitForCacheSync(stopCh, reqMgmt.plInformerSynced, reqMgmt.fsInformerSynced); !ok {
+		return fmt.Errorf("Never achieved initial sync")
+	}
+	klog.Info("Running reqmgmt config worker")
+	wait.Until(reqMgmt.runWorker, time.Second, stopCh)
+	klog.Info("Shutting down reqmgmt config worker")
+	return nil
+}
+
+func (reqMgmt *requestManagementSystem) runWorker() {
+	for reqMgmt.processNextWorkItem() {
+	}
+}
+
+func (reqMgmt *requestManagementSystem) processNextWorkItem() bool {
+	obj, shutdown := reqMgmt.configQueue.Get()
+	if shutdown {
+		return false
+	}
+
+	func(obj interface{}) {
+		defer reqMgmt.configQueue.Done(obj)
+		if !reqMgmt.syncOne() {
+			reqMgmt.configQueue.AddRateLimited(obj)
+		} else {
+			reqMgmt.configQueue.Forget(obj)
+		}
+	}(obj)
+
+	return true
+}
+
+// syncOne attempts to sync all the config for the reqmgmt filter.  It
+// either succeeds and returns `true` or logs an error and returns
+// `false`.
+func (reqMgmt *requestManagementSystem) syncOne() bool {
+	all := labels.Everything()
+	newPLs, err := reqMgmt.plLister.List(all)
+	if err != nil {
+		klog.Errorf("Unable to list PriorityLevelConfiguration objects: %s", err.Error())
+		return false
+	}
+	newFSs, err := reqMgmt.fsLister.List(all)
+	if err != nil {
+		klog.Errorf("Unable to list FlowSchema objects: %s", err.Error())
+		return false
+	}
+	reqMgmt.digestConfigObjects(newPLs, newFSs)
+	return true
+}
+
+func (reqMgmt *requestManagementSystem) digestConfigObjects(newPLs []*rmtypesv1a1.PriorityLevelConfiguration, newFSs []*rmtypesv1a1.FlowSchema) {
+	oldRMState := reqMgmt.curState.Load().(*requestManagementState)
+	var shareSum float64
+	newRMState := &requestManagementState{
+		priorityLevelStates: make(map[string]*priorityLevelState),
+	}
+	newlyQuiescent := make([]*priorityLevelState, 0)
+	for _, pl := range newPLs {
+		state := oldRMState.priorityLevelStates[pl.Name]
+		if state == nil {
+			state = &priorityLevelState{
+				config: pl.Spec,
+			}
+			newRMState.priorityLevelStates[pl.Name] = state
```

Comment:
Need to add to `shareSum` in this case too.

Reply:
nice catch, fixed